omod-dstu3/src/test/java/org/openmrs/module/fhir2/web/servlet/BaseFhirResourceProviderTest.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.web.servlet;
+
+import static org.springframework.http.HttpHeaders.ACCEPT;
+import static org.springframework.http.HttpHeaders.CONTENT_TYPE;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.validation.constraints.NotNull;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.stream.Collectors;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.parser.DataFormatException;
+import ca.uhn.fhir.parser.IParser;
+import ca.uhn.fhir.rest.api.RequestTypeEnum;
+import ca.uhn.fhir.rest.server.IResourceProvider;
+import ca.uhn.fhir.rest.server.interceptor.LoggingInterceptor;
+import lombok.SneakyThrows;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.OperationOutcome;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.openmrs.api.APIException;
+import org.openmrs.module.fhir2.FhirConstants;
+import org.openmrs.module.fhir2.api.impl.FhirGlobalPropertyServiceImpl;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.mock.web.MockServletConfig;
+import org.springframework.mock.web.MockServletContext;
+import org.springframework.transaction.annotation.Transactional;
+
+public abstract class BaseFhirResourceProviderTest<T extends IResourceProvider, U extends IBaseResource> {
```

Comment:
Instead of making a fork of this class, maybe it makes more sense to call the version here `BaseFhirR3ResourceProviderTest` and have it extend `BaseFhirResourceProviderTest` and then make the changes that we need here. This might involve changing `BaseFhirResourceProviderTest`, but that's ok.